Various improvements to the Terraform examples

andyrepton opened the pull request:
- Better README
- Removal of operator role prefix, it is now set to be the cluster name. This is to simplify the variables needed for users.
- Removal of account role prefix, it is now set to be the cluster name. This is to simplify the variables needed for users.
- A consistent experience across all terraform plans, they now will only ask for input if you would like to create a VPC and if you would like the cluster to be private or not
- Improved example tfvars files are now in the README
- var.cluster_name has been replaced with local.cluster_name, and if you do not provide a cluster_name it will now generate one for you in the format rosa-$random_5_digit_string. This helps for users who want to make multiple clusters automatically, and also removes another input needed for people getting started
- Removed the availability_zones variable and replaced it to automatically select three availability_zones based on the region provided, to remove another variable needed from the user
- Removed the ability to make single zone clusters as this adds a lot of complexity to the terraform plans. For users wanting to do this they can either write the tf code themselves or use the rosa command

Commit:
Cleaning up our TF files

andyrepton committed:
Merge branch 'main' into more_tf_fixes

paulczar commented on rosa_sts_managed_oidc/main.tf:
```diff
@@ -14,15 +14,26 @@
+
+locals {
+  # Extract availability zone names for the specified region
+  region_azs = [for zone in data.aws_availability_zones.available.names : format("%s", zone)]
```
what happens when a region has four AZs ?

Commit:
Various improvements to the Terraform examples, including:

- Better README
- Removal of operator role prefix, it is now set to be the cluster name. This is to simplify the variables needed for users.
- Removal of account role prefix, it is now set to be the cluster name. This is to simplify the variables needed for users.
- A consistent experience across all terraform plans, they now will only ask for input if you would like to create a VPC and if you would like the cluster to be private or not
- Improved example tfvars files are now in the README
- var.cluster_name has been replaced with local.cluster_name, and if you do not provide a cluster_name it will now generate one for you in the format rosa-$random_5_digit_string. This helps for users who want to make multiple clusters automatically, and also removes another input needed for people getting started
- Removed the availability_zones variable and replaced it to automatically select three availability_zones based on the region provided, to remove another variable needed from the user
- Removed the ability to make single zone clusters as this adds a lot of complexity to the terraform plans. For users wanting to do this they can either write the tf code themselves or use the rosa command
